### ISSUE TITLE
mpv + yt-dlp + sponsorblock + seek = no new video frames

### ISSUE THREAD

**Issue by Obegg** [mpv-player/mpv#13033]
### Important Information

Provide following Information:
- mpv version:
```
mpv v0.37.0-56-g67aa5684 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on Dec  2 2023 12:38:53
libplacebo version: v6.338.0-60-g795600a-dirty
FFmpeg version: N-112889-g5230257ea
FFmpeg library versions:
   libavutil       58.32.100
   libavcodec      60.35.100
   libavformat     60.18.100
   libswscale      7.6.100
   libavfilter     9.14.100
   libswresample   4.13.100
```
- Windows Version:
```
Edition	Windows 10 Enterprise LTSC
Version	21H2
Installed on	‎21/‎10/‎2023
OS build	19044.3693
Experience	Windows Feature Experience Pack 1000.19053.1000.0
```
- Source of the mpv binary:
https://github.com/zhongfly/mpv-winbuild/releases
- If known which version of mpv introduced the problem: No
- Possible screenshot or video of visual glitches: No

### Reproduction steps
While using mpv ([zhongfly](https://github.com/zhongfly/mpv-winbuild/releases)) + yt-dlp ([master](https://github.com/yt-dlp/yt-dlp-master-builds/releases)) + mpv_sponsorblock ([PR 71](https://github.com/po5/mpv_sponsorblock/blob/5a5aa4b9135f877c805e42c95dc69d5abe994b82/sponsorblock.lua) which skips `poi_highlight`) it seems like when seeking on [this YouTube video as example](https://www.youtube.com/watch?v=2QI88aLyaOs) (doesn't matter if it's auto seek or a manual seek) then yt-dlp will not show me any new frames from the video and I can only hear the audio track, or in some cases the stream itself will not load.

For testing purposes I've used mpv `portable_config` directory with the following commands on `mpv.conf`:
```
no-config
log-file="~~/yt-dlp-mpv-seek-issue.txt"
```
And added `sponsorblock` script with the following options on `sponsorblock.conf`:
```
categories=sponsor,selfpromo,interaction,intro,outro,preview,music_offtopic,filler,poi_highlight
skip_categories=sponsor,selfpromo,interaction,intro,outro,preview,music_offtopic,filler,poi_highlight
```

Note 1: Some information on the log file I've changed to `GitHub`.
Note 2: This issue has been bothering me for quite some time so it's not a new bug.
Note 3: The bug can be reproduced even without `sponsorblock` script, just a manual seek at the start of the stream to the middle of the stream.
Note 4: Created the same issue over at [yt-dlp issue tracker](https://github.com/yt-dlp/yt-dlp/issues/8705).

### Expected behavior
Show new video frames.

### Actual behavior
No new video frames, only audio.

### Log file
[yt-dlp-mpv-seek-issue.txt](https://github.com/mpv-player/mpv/files/13538107/yt-dlp-mpv-seek-issue.txt)

### Sample files
[portable_config.zip](https://github.com/mpv-player/mpv/files/13538106/portable_config.zip)
https://www.youtube.com/watch?v=2QI88aLyaOs

**Comment by mooreye** [comment]
I just rechecked and it seems that even without that sponsorblock LUA script, if you just click on a part of video on OSD that is not yet cached, only audio will be heard but no new frames will be displayed. So this seems to be an issue with mpv specifically.

**Comment by TheHardew** [comment]
For me the problem was ytdl-format set to prioritize the `vp09` codec.
```
617     mp4   1920x1080   60    │ ~754.85MiB 5259k m3u8  │ vp09.00.41.08 5259k video only
303     webm  1920x1080   60    │  325.23MiB 2267k https │ vp9           2267k video only          1080p60, webm_dash
```
It might seem that format 617 is much better, but after downloading them, it turns out they are exactly the same.
 
If that doesn't help, you can debug by running `mpv --no-config` to test whether you can seek. If it works, you can then run just `mpv` and try to enable user scripts and config options inside of mpv.conf until it breaks.

**Comment by guidocella** [comment]
`--cache=no` fixes seeking youtube videos for some reason. If you want to keep the cache you can use one of methods described in https://github.com/mpv-player/mpv/issues/8655#issuecomment-1199733728

**Comment by geextahslex** [comment]
@guidocella `cache=no` doesn't fix it in my case

**Comment by guidocella** [comment]
Using av1 instead of vp9 in https://www.youtube.com/watch?v=PgHZ4itvgVs fixes seeking rather than being fixed by `--cachie=no`.

**Comment by geextahslex** [comment]
But other videos run fine with VP9, there must be a some difference that triggers this issue

**Comment by guidocella** [comment]
I can reproduce being able to seek the AV1 stream and not the VP9 stream in ffplay so it's an ffmpeg issue.

**Comment by geextahslex** [comment]
As @Obegg is mentioning "frames" the only thing I can see is that the video that doesn't work has a weird FPS 29.97003.
Videos with VP9 60/30/25 FPS work well
 ![fps](https://github.com/mpv-player/mpv/assets/139760266/63aad938-2b6c-49c2-a8be-59b64a5ddcd6)

**Comment by guidocella** [comment]
`ytdl-format=(bestvideo[vcodec^=av01]/bestvideo)+bestaudio/best` circumvents the issue.

**Comment by geextahslex** [comment]
>To be honest I am quite happy I am not the only one experiencing this

@Obegg I felt the same as I wasn't sure if to make the issue in the first place. 

>`ytdl-format=(bestvideo[vcodec^=av01]/bestvideo)+bestaudio/best` circumvents the issue

@guidocella A workaround is always nice but VP9 and VP9 2 is youtubes workhorse. Going forward it would be nice to have this issue solved

**Comment by WPMGPRoSToTeMa** [comment]
> @guidocella A workaround is always nice but VP9 and VP9 2 is youtubes workhorse. Going forward it would be nice to have this issue solved
> 
> Would it be possible to only switch to AV01 if the faulty framerate is detected?

As far as I can see it is broken only for `vp09.AB.CD.EF` (e.g. `vp09.00.40.08`) and this codec is not always preferred by `yt-dlp`, e.g. I also see `vp9` is preferred in some cases. I use the following selector to filter out the buggy codec:
`ytdl-format=bv[vcodec!^=vp09.]+ba/b`

**Comment by guidocella** [comment]
https://github.com/yt-dlp/yt-dlp/issues/5629

**Comment by geextahslex** [comment]
I'm not sure but for me I have a better performance with VP9 than with AV01
AV01
![av01](https://github.com/mpv-player/mpv/assets/139760266/a24bead5-2413-431b-80a3-bfd4854a9371)
VP9
![vp9](https://github.com/mpv-player/mpv/assets/139760266/92580f0f-101a-44ff-9331-cbff263df867)

**Comment by SnnUntz** [comment]
`ytdl-raw-options = format-sort="proto"`

**Comment by geextahslex** [comment]
>ytdl-raw-options = format-sort="proto"

what does it mean/do? ^^'

**Comment by SnnUntz** [comment]
> > ytdl-raw-options = format-sort="proto"
> 
> what does it mean/do? ^^'

# Download best video available via direct link over HTTP/HTTPS protocol,
# or the best video available via any protocol if there is no such video
$ yt-dlp -f "(bv*+ba/b)[protocol^=http][protocol!*=dash] / (bv*+ba/b)"

# Download best video available via the best protocol
# (https/ftps > http/ftp > m3u8_native > m3u8 > http_dash_segments ...)
$ yt-dlp -S "proto"

**Comment by geextahslex** [comment]
@SnnUntz 
>These can be used to solve the issue.
`ytdl-raw-options = format-sort="proto"`
or
`ytdl-raw-options = extractor-args="youtube:skip=hls"`
or
`ytdl-format = "(bv*+ba/b)[protocol^=http][protocol!*=dash] / (bv*+ba/b)"`

Thats amazing, all of them work. But what does it change? Why is it working like this? ^^'

**Comment by SnnUntz** [comment]
> Thats amazing, all of them work. But what does it change? Why is it working like this? ^^'

The bug occurs when using the m3u8 protocol.

**Comment by geextahslex** [comment]
@SnnUntz 
>Then this should work too: `ytdl-format = bv*[vcodec!*=vp09]+ba/b`

Yes this works too, but you loose the ability to pick your resolution with something like `ytdl-format=bestvideo[height<=?1440]+bestaudio/best`

**Comment by SnnUntz** [comment]
After some research I found that it was an ffmpeg issue.
https://trac.ffmpeg.org/ticket/7359
https://trac.ffmpeg.org/ticket/11006

**Comment by WPMGPRoSToTeMa** [comment]
> Then this should work too: `ytdl-format = bv*[vcodec!*=vp09]+ba/b`

Yeah, I mentioned a similar fix previously https://github.com/mpv-player/mpv/issues/13033#issuecomment-2198678823.

> Yes this works too, but you loose the ability to pick your resolution with something like `ytdl-format=bestvideo[height<=?1440]+bestaudio/best`

If you use `uosc` like me, then it can be fixed in their .lua code. But yeah, this is exactly the reason I currently use `ytdl-raw-options = extractor-args="youtube:skip=hls"`.

**Comment by SnnUntz** [comment]
> > Yes this works too, but you loose the ability to pick your resolution with something like `ytdl-format=bestvideo[height<=?1440]+bestaudio/best`
> 
> If you use `uosc` like me, then it can be fixed in their .lua code. But yeah, this is exactly the reason I currently use `ytdl-raw-options = extractor-args="youtube:skip=hls"`.

You can also use it this way: `bestvideo*[vcodec!*=vp09][height<=?1440]+bestaudio/best`

**Comment by ValeZAA** [comment]
> After some research I found that this is an ffmpeg issue. https://trac.ffmpeg.org/ticket/7359 https://trac.ffmpeg.org/ticket/11006

That is fixed btw

**Comment by Dudemanguy** [comment]
Allegedly fixed upstream.

**Comment by Hund** [comment]
Do we know which version of ffmpeg this fix will be shipped with? This bug is getting old now. Pun intended.
